src/library/Statistics.nim

```diff
@@ -134,24 +134,25 @@ proc defineSymbols*() =
         example     = """
             print median [2 4 5 6 7 2 3]
             ; 6
-            
+
             print median [1 5 2 3 4 7 9 8]
             ; 3.5
         """:
             #=======================================================
-            if x.a.len==0: 
+            if x.a.len==0:
                 push(VNULL)
             else:
+                let sorted = x.a.sorted(order = SortOrder.Ascending)
                 let first = x.a[(x.a.len-1) div 2]
                 let second = x.a[((x.a.len-1) div 2)+1]
 
                 if x.a.len mod 2 == 1:
```

Comment:
```suggestion
                if sorted.len mod 2 == 1:
```

Since we now have `sorted`, better use this one. (OK, it's a bit of an overkill to use for `.len` as well, but it looks cleaner I guess). In the case, of `first` and `second`, it's a must, since x.a is not sorted 😉

Reply:
Mea culpa, I forgot to update the variables, since I committed the code from Github's fast commit.